lib/src/benchmark_base.dart

```diff
@@ -35,32 +35,55 @@ class BenchmarkBase {
 
   // Measures the score for this benchmark by executing it repeatedly until
   // time minimum has been reached.
-  static double measureFor(void Function() f, int minimumMillis) {
-    var minimumMicros = minimumMillis * 1000;
-    var iter = 0;
-    var watch = Stopwatch();
-    watch.start();
+  static _Measurement _measureForImpl(
+      void Function() f, int minimumMillis, int initialIter) {
+    final minimumMicros = minimumMillis * 1000;
+    var iter = initialIter;
     var elapsed = 0;
-    while (elapsed < minimumMicros) {
-      f();
+    final watch = Stopwatch()..start();
+    while (true) {
+      watch.reset();
+      for (var i = 0; i < iter; i++) {
+        f();
+      }
       elapsed = watch.elapsedMicroseconds;
-      iter++;
+      if (elapsed >= minimumMicros) {
+        return _Measurement(elapsed, iter);
+      }
+      if (elapsed == 0) {
+        iter *= 1000;
+      } else {
+        iter = (iter * math.max(minimumMicros / elapsed, 2.0)).ceil();
+      }
     }
-    return elapsed / iter;
   }
 
+  // Measures the score for this benchmark by executing it repeatedly until
+  // time minimum has been reached.
```

Comment:
Consider making these triple `/// ` comments so they show up in the API docs (also for the other methods)